src/GitHub.UI/Helpers/LoadingResourceDictionary.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.IO;
+using System.Windows;
+using System.Reflection;
+using System.Diagnostics;
+using System.Collections.Generic;
+
+namespace GitHub
+{
+    public class LoadingResourceDictionary : ResourceDictionary
+    {
+        static Dictionary<string, Assembly> assemblyDicts = new Dictionary<string, Assembly>();
+
+        public new Uri Source
```

Comment:
This was getting messy, so I thought I'd have a crack at finding out why the XAML designer doesn't like `SharedDictionaryManager`. With any luck `XAML_DESIGNER` will no longer be needed! See here #945.

I've added the `#ifdef !XAML_DESIGNER` anyway as part of this PR.